### PR TITLE
feat: new api view for CSV formatted indexed data

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
## Description

We want to enable users to download faceted algolia results to a CSV file. This PR adds a backend view that enables us to retrieve and format the data needed for the download.

## Ticket Link

Link to the associated ticket
[Jira- CSV download button ticket](https://openedx.atlassian.net/browse/ENT-5084)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/views.py [hunk 1]
```diff
@@ -164,6 +169,148 @@ def contains_content_items(self, request, uuid, course_run_ids, program_uuids, *
...
+
+        pacing_type = hit['advertised_course_run']['pacing_type']
+        csv_row.append(pacing_type) if pacing_type else csv_row.append('No pacing type')
+
+        csv_row.append(hit.get('level_type', 'No level_type'))
+
+        csv_row.append(hit['advertised_course_run'].get('first_enrollable_paid_seat_price', 'No price'))
+
+        # Min, max effort and length TBD
+        csv_row.append('TBD')
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views.py]: TBD - what will go in here?

File: enterprise_catalog/apps/api/v1/views.py [hunk 2]
```diff
@@ -164,6 +169,148 @@ def contains_content_items(self, request, uuid, course_run_ids, program_uuids, *
...
+                combined_facets.append(f'{facet_name}:{facet_value}')
+            facet_filters.append(combined_facets)
+
+        # Algolia search will only retrieve all results if you query by empty string.
+        algolia_hits = algolia_client.algolia_index.browse_objects({
+            'query': '',
+            'facetFilters': facet_filters,
+            'attributesToRetrieve': self.algolia_attributes_to_retrieve
+        })
+        with StringIO() as file:
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views.py:336]: 🙌

**Review by johnnagro — APPROVED** [review]
i assume those TBD placeholders will get swapped before we put the UI button in place?

File: enterprise_catalog/apps/api/v1/views.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/views.py]: We don't have a value here in Algolia so we need to figure out how we're going to rectify the missing data. Ultimately a convo with Mariana

**Review by johnnagro — APPROVED** [review]
⛵

File: enterprise_catalog/apps/catalog/algolia_utils.py [hunk 3]
```diff
@@ -637,6 +637,7 @@ def get_advertised_course_run(course):
     full_course_run = _get_course_run_by_uuid(course, course.get('advertised_course_run_uuid'))
     if full_course_run is None:
         return None
+    # upgrade_deadline is recorded in EPOCH time
```
Comment (kiram15) [inline enterprise_catalog/apps/catalog/algolia_utils.py:640]: Thank you for adding this in there!

**Review by kiram15 — APPROVED** [review]
LGTM!

**Pull request merged**